ccmlib/scylla_docker_cluster.py

```diff
@@ -27,6 +28,78 @@ def create_node(self, name, auto_bootstrap, thrift_interface,
+        node_list = [node.name for node in list(self.nodes.values())]
+        seed_list = [node.name for node in self.seeds]
+        filename = os.path.join(self.get_path(), 'cluster.conf')
+        docker_image = self.docker_image
+        with open(filename, 'w') as f:
+            yaml.safe_dump({
+                'name': self.name,
+                'nodes': node_list,
+                'seeds': seed_list,
+                'partitioner': self.partitioner,
+                'config_options': self._config_options,
+                'id': self.id,
+                'ipprefix': self.ipprefix,
+                'docker_image': docker_image
+            }, f)
+
+    def populate(self, nodes, debug=False, tokens=None, use_vnodes=False, ipprefix=None, ipformat=None):
+        node_count = nodes
+        dcs = []
+        if isinstance(nodes, list):
+            node_count = 0
+            i = 0
+            for c in nodes:
+                i = i + 1
+                node_count = node_count + c
+                for x in range(0, c):
+                    dcs.append('dc%d' % i)
+
+        if node_count < 1:
+            raise common.ArgumentError('invalid node count %s' % nodes)
+
+        for i in range(1, node_count + 1):
+            if 'node%s' % i in list(self.nodes.values()):
+                raise common.ArgumentError('Cannot create existing node node%s' % i)
+
+        for i in range(1, node_count + 1):
+            tk = None
+            if tokens is not None and i - 1 < len(tokens):
+                tk = tokens[i - 1]
+            dc = dcs[i - 1] if i - 1 < len(dcs) else None
+            self.new_node(i, debug=debug, initial_token=tk, data_center=dc)
+            self._update_config()
+        return self
+
+    def new_node(self, i, auto_bootstrap=False, debug=False, initial_token=None, add_node=True, is_seed=True, data_center=None):
+        node = self.create_node(name='node{}'.format(i),
+                                auto_bootstrap=auto_bootstrap,
+                                thrift_interface=self.get_thrift_interface(i),
+                                storage_interface=self.get_storage_interface(i),
+                                jmx_port=str(self.get_node_jmx_port(i)),
+                                remote_debug_port=str(self.get_debug_port(i) if debug else 0),
+                                initial_token=initial_token)
+        if add_node:
+            self.add(node, is_seed=is_seed, data_center=data_center)
+        return node
+
+    def add(self, node, is_seed, data_center=None):
+        if node.name in self.nodes:
+            raise common.ArgumentError('Cannot create existing node %s' % node.name)
+        self.nodes[node.name] = node
+        if is_seed:
+            self.seeds.append(node)
+        self._update_config()
+        node.data_center = data_center
+
+        if data_center is not None:
+            self.__update_topology_files()
+        node.update_yaml()
+        node.create_docker()
```

Comment:
why the `create_docker()` is here ?

Reply:
because the function `add` must add the nodes to the cluster. When i create a new cluster with `-n X` I expect a cluster to be created with `X` nodes in it

Reply:
`add` isn't starting the nodes, just create the `Node` object and connect it to the `Cluster` object